### ISSUE TITLE
Get References doesn't find symbols from template instantiation -- e.g. `std::make_uinque<Thing>()`

### ISSUE THREAD

**Issue by lanza** [MaskRay/ccls#620]
As mentioned in the title -- the LSP method GetReferences does not find the reference to `Thing::Thing()` as invoked via `std::make_unique<Thing>()` when it's defined in a different file.

```
// Thing.hpp
class Thing {
public:
  Thing::Thing() {}
};
```
```
// main.cpp
#include <memory>
#include "Thing.hpp"
int main() {
  auto t = std::make_unique<Thing>();
}
```

This is also fails given a different TU (e.g. if Thing.cpp was defined and the constructor defined there instead). But it works given the constructor defined in the same TU as the `std::make_unique` invocation.

**Comment by MaskRay** [comment]
Yeah, finding references on the constructor should be able to find `std::make_unique` and potentially other common smart constructors. This may require some heuristics.
